docs/dev/guidelines/database.rst

```diff
@@ -0,0 +1,235 @@
+
+ .. code-block:: java
+
+    // Result.java
+    @OneToMany(mappedBy = "result", cascade = CascadeType.ALL, orphanRemoval = true)
+    @OrderColumn
+    @JsonIgnoreProperties(value = "result", allowSetters = true)
+    @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+    @JsonView(QuizView.Before.class)
+    private List<Feedback> feedbacks = new ArrayList<>();
+
+
+* ``CascadeType.PERSIST`` When persisting a parent entity, it also persists the child entities held in its fields. This cascade rule is helpful for relationships where the parent acts as a *container* to the child entity. If you do not use this, you have to ensure that you persist the child entity first, otherwise an error will be thrown. Example: The code below propagates the **persist** operation from parent ``AnswerCounter`` to child ``AnswerOption``. When an ``AnswerCounter`` is persisted, its ``AnswerOption`` is persisted as well. 
+
+ .. code-block:: java
+
+    // AnswerCounter.java
+    @OneToOne(cascade = { CascadeType.PERSIST })
+    @JoinColumn(unique = true)
+    private AnswerOption answer;
+
+
+* ``CascadeType.MERGE`` If you merge the source entity (saved/updated/synchronized)  to the database, the merge is cascaded to the target of the association. This rule applies to existing objects only. Use this type to always merge/synchronize the existing data in the table with the data in the object. Example below: whenever we merge a ``Result`` to the database, i.e. save the changes on the object, the ``Assessor`` object is also merged/saved. 
+
+ .. code-block:: java
+
+    // Result.java
+    @OneToOne(cascade = CascadeType.MERGE, fetch = FetchType.LAZY)
+    @JoinColumn(unique = false)
+    private User assessor;
+
+
+* ``CascadeType.REMOVE`` If the source entity is removed, the target of the association is also removed. Example below: propagates **remove** operation from parent ``Submission`` to child ``Result``. When a ``Submission`` is deleted, also the ``Result`` is deleted.
+
+ .. code-block:: java
+
+    // Submission.java
+    @OneToOne(mappedBy = "submission", fetch = FetchType.LAZY, cascade = CascadeType.REMOVE)
+    @JsonIgnoreProperties({ "submission", "participation" })
+    @JoinColumn(unique = true)
+    private Result result;
+ 
+
+* ``CascadeType.REFRESH`` If the source entity is refreshed, it cascades the refresh to the target of the association. This is used to refresh the data in the object and its associations. This is useful for cases where there is a change which needs to be synchronized FROM the database.
+
+Not used in Artemis yet
+
+
+Best Practices
+===============
+* If you want to create a ``@OneToMany`` relationship or ``@ManyToMany`` relationship, first think about if it is important for the association to be ordered. If you do not need to association to be ordered, then always go for a ``Set`` instead of ``List``. If you are unsure, start with a ``Set``. 
```

Comment:
```suggestion
* If you want to create a ``@OneToMany`` relationship or ``@ManyToMany`` relationship, first think about if it is important for the association to be ordered. If you do not need the association to be ordered, then always go for a ``Set`` instead of ``List``. If you are unsure, start with a ``Set``. 
```